### DISCUSSION TITLE
Creating arrays of objects in server

### DISCUSSION THREAD

**Discussion by 111-jon** [open62541pp/open62541pp#300] [Q&A]
I have a server implementation working, serving from open62541pp on Linux to Siemens HMI.

I need to be able to share data from many objects managed by my server to the HMI so that the appear as an array of object types which can be browsed by numerical index.  In the C++, we'd do e.g. wanted_value = object[i].that_value, where object is e.g. struct { int this_value, bool that_value, float another_value}

Essentially, I want to be able to share members from multiple instances of the same C++ object to the HMI.

This is probably very straightforward, but I am unable to find any clues so far.  A bullet point guide to implementing arrays of objects based around the open62541p tools would be very helpful.  Please help!

**Comment by lukasberbuer** [comment]
Hi @111-jon, 

I hope I understand you correctly. Every node has a unique id, a `NodeId`. In the end, you just need to save this id and can always reference the object/variables. You can create a `opcua::Node` object by constructing it from the `Server` instance and the `NodeId`.

If you don't want to write the values to the server whenevery they change, you can use other mechanisms like [value callbacks](https://github.com/open62541pp/open62541pp/blob/master/examples/server_valuecallback.cpp) or [value data sources](https://github.com/open62541pp/open62541pp/blob/master/examples/server_datasource.cpp).
